src/h5web/explorer/utils.ts

```diff
@@ -87,17 +88,32 @@ export function buildTree(
   });
 }
 
+function findRoot(entity: MyHDF5Group): MyHDF5Metadata {
+  const { parent } = entity;
+  return parent ? findRoot(parent) : entity;
+}
```

Comment:
Moved up from below and simplified to only accept groups.

Reply:
I would have kept `MyHDF5Entity` as the arg type as it might be useful to find the root from any entity.

Reply:
Arf, I remember now why I changed it. If I change the arg to `MyHDF5Entity`, I need to change the return type to `MyHDF5Entity` as well. That's because a dataset for instance may not have a parent, and the dataset itself is returned in this case. However, this doesn't seem to affect the `getEntityAtPath` function, so the change makes sense. We can always add an override with `MyHDF5Group` later on if we need to.

Reply:
I would argue that if you can't follow the parents up to the root group, it should return `undefined`. But then, it raises other problems...

Reply:
Well, it's arguable since the dataset is, in that case, the root of the structure! 😄

Reply:
That's what I had initially, but it was all a bit more pointlessly complicated, I thought.

Reply:
Ha, perhaps we don't have the same view on the signification of `root`. 

For me, it should be the root group of the file and nothing else. But as you say, it is arguable since we are dealing here with the tree structure and not the file structure itself.